4-WebApp-your-API/4-1-MyOrg/TodoListService/Startup.cs

```diff
@@ -26,11 +32,51 @@ public void ConfigureServices(IServiceCollection services)
             // By default, the claims mapping will map claim names in the old format to accommodate older SAML applications.
             // 'http://schemas.microsoft.com/ws/2008/06/identity/claims/role' instead of 'roles'
             // This flag ensures that the ClaimsIdentity claims collection will be built from the claims in the token
-            // JwtSecurityTokenHandler.DefaultMapInboundClaims = false;
+            JwtSecurityTokenHandler.DefaultMapInboundClaims = false;
 
             // Adds Microsoft Identity platform (AAD v2.0) support to protect this Api
             services.AddMicrosoftIdentityWebApiAuthentication(Configuration);
 
+
+            //// Comment the lines of code above and uncomment the following section if you would like to limit calls to this API to just a set of client apps
+            //// The following is an example of extended token validation
+            /**
+            * The example below can be used do extended token validation and check for additional claims, such as:
+            * -check if the caller's tenant is in the allowed tenants list via the 'tid' claim (for multi-tenant applications)
+            *  -check if the caller's account is homed or guest via the 'acct' optional claim
+                * -check if the caller belongs to right roles or groups via the 'roles' or 'groups' claim, respectively
+                *
+                * For more information, visit: https://docs.microsoft.com/azure/active-directory/develop/access-tokens#validate-the-user-has-permission-to-access-this-data
+            **/
+
+
+
+                       //services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
+                       //  .AddMicrosoftIdentityWebApi(options =>
+                       //      {
+                       //          Configuration.Bind("AzureAd", options);
+                       //          options.Events = new JwtBearerEvents();
+                       //          options.Events.OnTokenValidated = async context =>
+                       //          {
+                       //              string[] allowedClientApps =
+                       //              {
+                       //                  /* list of client ids to allow */
+                       //              };
+                       //              string clientappId = context?.Principal?.Claims
+                       //                  .FirstOrDefault(x => x.Type == "azp" || x.Type == "appid")?.Value;
+
+                       //              if (!allowedClientApps.Contains(clientappId))
+                       //              {
+                       //                  throw new HttpResponseException(HttpStatusCode.Unauthorized);
+                       //              }
+                       //          }
+                       //      }
+                       //  );
```

Comment:
this is still not the latest code and needs some looking and testing to ensure we ship the working bits